scripts/gha/integration_testing/xcodebuild.py

```diff
@@ -0,0 +1,114 @@
+get_args_for_build() step and then use generate_unsigned_ipa() to package
+the .app to .ipa.
+
+"""
+
+import os
+import shutil
+
+
+def get_args_for_build(path, scheme, output_dir, ios_sdk, configuration):
+  """Constructs subprocess args for an unsigned xcode build.
+
+  Args:
+    path (str): Full path to the project or workspace to build. Must end in
+        either .xcodeproj or .xcworkspace.
+    scheme (str): Name of the scheme to build.
+    output_dir (str): Directory for the resulting build artifacts. Will be
+        created if it doesn't already exist.
+    ios_sdk (str): Where this build will be run: device or simulator.
+    configuration (str): Value for the -configuration flag.
+
+  Returns:
+    Sequence of strings, corresponding to valid args for a subprocess call.
+
+  """
+  args = [
+      "xcodebuild",
+      "-sdk", _get_ios_env_from_target(ios_sdk),
+      "-scheme", scheme,
+      "-configuration", configuration,
+      "-quiet",
+      "BUILD_DIR=" + output_dir
+  ]
+
+  if ios_sdk == "device":
+    args.extend(['CODE_SIGN_IDENTITY=""',
+      "CODE_SIGNING_REQUIRED=NO",
+      "CODE_SIGNING_ALLOWED=NO"])
+  elif ios_sdk == "simulator":
+    args.extend(['-arch', "x86_64"])
+
+  if not path:
+    raise ValueError("Must supply a path.")
+  if path.endswith(".xcworkspace"):
+    args.extend(("-workspace", path))
+  elif path.endswith(".xcodeproj"):
+    args.extend(("-project", path))
+  else:
+    raise ValueError("Path must end with .xcworkspace or .xcodeproj: %s" % path)
+  return args
+
+
+def _get_ios_env_from_target(ios_sdk):
+  """Return a value for the -sdk flag based on the target (device/simulator)."""
+  if ios_sdk == "device":
+    return "iphoneos"
+  elif ios_sdk == "simulator":
+    return "iphonesimulator"
+  else:
+    raise ValueError("Unrecognized ios_sdk: %s" % ios_sdk)
+
+
+def generate_unsigned_ipa(output_dir, configuration):
+  """Creates an unsigned .ipa from an existing .app.
+
+  Args:
+    output_dir (str): Same value as get_args_for_build. The generated unsigned
+      .ipa will be placed within the subdirectory "<configuration>-iphoneos".
+    configuration (str): Same value as get_args_for_build.
+    app_name (str): Name of the .app mobile application (without extension).
+  """
+  iphone_build_dir = os.path.join(output_dir, configuration + "-iphoneos")
+  payload_path = os.path.join(iphone_build_dir, "Payload")
+  for _, directories, _ in os.walk(iphone_build_dir):
+    for directory in directories:
+      if directory.endswith('.app'):
+        app_name = os.path.splitext(directory)[0]
+        app_path = os.path.join(iphone_build_dir, app_name + ".app")
+        ipa_path = os.path.join(iphone_build_dir, app_name + ".ipa")
+        os.mkdir(payload_path)
+        shutil.move(app_path, payload_path)
+        shutil.make_archive(
+            payload_path, "zip", root_dir=iphone_build_dir, base_dir="Payload")
+        shutil.move("%s.%s"%(payload_path, "zip"), ipa_path)
```

Comment:
[f-string](https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals) appears to be the modern way for formatting strings. I think they are more readable than `str.format()` and `%` operator. It's up to you.

Reply:
`f-string` is a new feature in python 3.6. Our script used to support python 2. But I think that it's worth doing this cleaning for the python scripts.